packages/utils/telemetry-utils/src/batchedTelemetryHelper.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type { IDisposable } from "@fluidframework/core-interfaces";
+
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+
+/**
+ * TODO
+ */
+interface ICustomDataMap {
```

Comment:
I think this is equivalent, and we tend to prefer the `Record<>` approach.
```suggestion
type ICustomDataMap = Record<string, number>;
```